### ISSUE TITLE
Disable Log Upload to Azure DevOps

### ISSUE THREAD

**Issue by jonathaneckman** [microsoft/azure-pipelines-agent#2608]
Our team would like to continue using Azure DevOps Service, but are facing a security requirement we cannot get around. In select environments, we do not want to upload deployment logs to Azure DevOps. Is there currently a way to disable this upload? If not, is this a feature you'd consider? If we can disable them we would then next look at publishing them to an internally managed Azure service, such as Log Analytics.

**Comment by vtbassmatt** [comment]
@jonathaneckman can you describe more about what's behind this requirement? Besides the logs, a lot of other information about the pipeline run gets stored in Azure DevOps. I think it's unlikely we'd be able to take this work as stated, but maybe there's something else we can do to meet the requirement.

**Comment by jonathaneckman** [comment]
We were recently trying to decide between Azure DevOps Service and Server. Service was preferred by all, but we had some strict requirements around what data was stored in an Azure commercial region. We found a solution for everything except for the deployment logs. 

We have a CI/CD pipeline deploying a complex infrastructure-as-code solution. Our environment is uniquely strict so things like Azure resource ids, URLs, IP addresses, and usernames cannot be stored in the Azure regions Azure DevOps Service is currently hosted in. While we tuck most of this away in keyvault, we were uncomfortable with how easy it would be for some of this info to slip out so decided to host Azure DevOps ourselves. The info Azure DevOps needs to power the pipeline was not a concern. Its the output of the tasks from our own code and task configuration. 

While we have already moved on, here are some ideas that may help us drop Server and go back to Service:

- Host Azure DevOps in a Azure Government regions (preferred)
- Allow task log output to be disabled at the Environment level (yaml pipelines), not at the code level.
- Let us choose where task log output is stored. Ex: on the build agent or in Azure Storage account.
